src/config_validator.py

```
@@ -0,0 +1,359 @@
+        content = file_path.read_text(encoding='utf-8')
+
+        if file_path.suffix == '.md':
+            # Extract YAML from markdown
+            yaml_blocks = extract_yaml_from_markdown(content)
+            results['yaml_blocks'] = len(yaml_blocks)
+
+            for i, yaml_content in enumerate(yaml_blocks):
+                try:
+                    # Handle multi-document YAML blocks
+                    documents = list(yaml.safe_load_all(yaml_content))
+                    for doc_idx, config in enumerate(documents):
```

Ruff (0.13.0)

Loop control variable `doc_idx` not used within loop body

Rename unused `doc_idx` to `_doc_idx`

(B007)